### PR TITLE
Split pandaState into peripheralState and pandaState

### PR DESCRIPTION
**Pull request by pd0wm**

### CODE REVIEW

File: selfdrive/thermald/thermald.py [hunk 1]
```diff
@@ -203,30 +201,29 @@ def thermald_thread():
...
         if no_panda_cnt > DISCONNECT_TIMEOUT / DT_TRML:
           if startup_conditions["ignition"]:
             cloudlog.error("Lost panda connection while onroad")
           startup_conditions["ignition"] = False
       else:
         no_panda_cnt = 0
-        startup_conditions["ignition"] = pandaState.pandaState.ignitionLine or pandaState.pandaState.ignitionCan
+        startup_conditions["ignition"] = pandaState.ignitionLine or pandaState.ignitionCan
 
-      startup_conditions["hardware_supported"] = pandaState.pandaState.pandaType not in [log.PandaState.PandaType.whitePanda,
```
Comment (pd0wm) [inline selfdrive/thermald/thermald.py:223]: Already asserted here: https://github.com/commaai/openpilot/blob/master/selfdrive/boardd/panda.cc#L76

File: selfdrive/boardd/boardd.cc [hunk 2]
```diff
@@ -392,8 +376,47 @@ void panda_state_thread(Panda *&panda, bool spoofing_started) {
   }
 }
 
-void hardware_control_thread(Panda *panda) {
-  LOGD("start hardware control thread");
+void peripheral_state_thread(Panda *panda) {
```
Comment (adeebshihadeh) [inline selfdrive/boardd/boardd.cc:379]: this should stay in the panda_state_thread. we don't really want the scheduler to handle this for us

File: selfdrive/boardd/boardd.cc [hunk 3]
```diff
@@ -392,8 +376,47 @@ void panda_state_thread(Panda *&panda, bool spoofing_started) {
...
+
+    auto ps = evt.initPeripheralState();
+    ps.setPandaType(panda->hw_type);
+
+    if (Hardware::TICI()) {
+      double read_time = millis_since_boot();
+      ps.setVoltage(std::atoi(util::read_file("/sys/class/hwmon/hwmon1/in1_input").c_str()));
+      ps.setCurrent(std::atoi(util::read_file("/sys/class/hwmon/hwmon1/curr1_input").c_str()));
+      read_time = millis_since_boot() - read_time;
+      if (read_time > 50) {
```
Comment (adeebshihadeh) [inline selfdrive/boardd/boardd.cc:400]: side note: I checked the cloudlogs and this actually lags heavily sometimes

File: selfdrive/thermald/thermald.py [hunk 4]
```diff
@@ -203,30 +202,30 @@ def thermald_thread():
     params.put_bool("BootedOnroad", True)
 
   while 1:
-    pandaState = messaging.recv_sock(pandaState_sock, wait=True)
+    sm.update(pandaState_timeout)
```
Comment (adeebshihadeh) [inline selfdrive/thermald/thermald.py:205]: hmm, probably doesn't change much, but do we really want to conflate here?

File: selfdrive/thermald/tests/test_power_monitoring.py [hunk 5]
```diff
@@ -37,20 +37,17 @@ def setUp(self):
...
 
-  def mock_pandaState(self, ignition, hw_type, car_voltage=12, harness_status=log.PandaState.HarnessStatus.normal):
-    pandaState = messaging.new_message('pandaState')
-    pandaState.pandaState.pandaType = hw_type
-    pandaState.pandaState.voltage = car_voltage * 1e3
-    pandaState.pandaState.ignitionLine = ignition
-    pandaState.pandaState.ignitionCan = False
-    pandaState.pandaState.harnessStatus = harness_status
-    return pandaState
+  def mock_peripheralState(self, hw_type, car_voltage=12,):
```
Comment (adeebshihadeh) [inline selfdrive/thermald/tests/test_power_monitoring.py:40]:
```suggestion
  def mock_peripheralState(self, hw_type, car_voltage=12):
```

File: selfdrive/thermald/tests/test_power_monitoring.py [hunk 6]
```diff
@@ -37,20 +37,17 @@ def setUp(self):
...
+    peripheralState = messaging.new_message('peripheralState')
+    peripheralState.peripheralState.pandaType = hw_type
+    peripheralState.peripheralState.voltage = car_voltage * 1e3
+    return peripheralState.peripheralState
```
Comment (adeebshihadeh) [inline selfdrive/thermald/tests/test_power_monitoring.py:44]:
```suggestion
    ps = messaging.new_message('peripheralState').peripheralState
    ps.pandaType = hw_type
    ps.voltage = car_voltage * 1e3
    return ps
```

File: selfdrive/thermald/thermald.py (hunk 4 above)
Comment (pd0wm) [reply selfdrive/thermald/thermald.py:205]: I'll change it back to a regular socket, then I also don't have to do the other check

**Pull request merged**